api/types/kustomization.go

```diff
@@ -301,6 +301,27 @@ func (k *Kustomization) FixKustomizationPreMarshalling(fSys filesys.FileSystem)
 	return nil
 }
 
+func (k *Kustomization) CheckEmpty() error {
```

Comment:
Am I right to think that we want to throw an error on any file that does not contain any transformers? The code below effectively excludes files that specifically contain only Kustomization type metadata, but I suspect it would actually allow fully empty files or files with partial type meta (which is currently allowed, for better or for worse). I suggest changing the tests to be against raw yaml input rather than struct input to demonstrate the various cases more thoroughly. I also suggest trying the opposite approach for the comparison: deep copy the input Kustomization, clear its type meta, and then compare it to a completely empty object (in object form, not as yaml). LMK what you think of that approach.

Reply:
I agree with your opinions.
I fixed these codes.